### DISCUSSION TITLE
Example using Renderer.Screenshot

### DISCUSSION THREAD

**Discussion by jeffsager** [StereoKit/StereoKit#562] [Q&A] [answered]
Hi,
Is there any sample code for renderer.screenshot?  I have tried to get it to work on Hololens and think I'm handling the from and at incorrectly.
Thanks,
Jeff

**Comment by maluoi** [comment]
There's a couple uses of this in the StereoKitTest project, [like over here](https://github.com/StereoKit/StereoKit/blob/master/Examples/StereoKitTest/DebugToolWindow.cs#L134)! I don't think I have nice "sample" stuff for it yet though, hopefully that's enough :)

**Comment by jeffsager** [comment, marked as answer]
That is perfect, thank you!

**Comment by jeffsager** [comment]
I figured Renderer.Screenshot out (with your sample) but have a quick question, is the screenshot supposed to only include the holograms on a black background? I'm on Hololens 2. I can capture what the Hololens "sees" as an image using the CameraCaptureUI API but the users hate Microsoft's camera interface.  I can capture "automatically" using MediaCapture API but it does not pick up the holograms.
I am looking for a "frictionless" way to capture what is visible on the Hololens 2.
Any ideas would be greatly appreciated.. I'd be happy to share the code for CameraCaptureUI and MediaCapture if it would help anyone.
Thanks,
Jeff
I can start a new discussion if that would be best.

**Reply by maluoi** [reply]
`Renderer.Screeshot` can take a picture from any position and direction, which means compositing with the real world is not exactly feasible to expect there!

The `MediaCapture` API is the correct way to go, and it _should_ composite with the holograms. I've definitely seen OS updates break this functionality in the past to some degree or another, but you can verify that it's working by taking a screenshot using the OS screenshot capability, or using the live streaming feature. If either of those work, then `MediaCapture` should also work. It's possible there may be a setting somewhere in the API to specify compositing or not, but I don't recall it off the top of my head!

**Comment by jeffsager** [comment]
Thanks for the answer.  Finally figured it out. I'm attaching the code for the VideoMRCSettings and calling the media capture. Calling media capture and not adding the video affect produces the image minus the holograms. I  apologize in advance that this code is a little messy.  I kind of threw the kitchen sink at it.  I'll be cleaning it up over the next few days.

```csharp
public class VideoMRCSettings : IVideoEffectDefinition
{
    public string ActivatableClassId
    {
        get
        {
            return "Windows.Media.MixedRealityCapture.MixedRealityCaptureVideoEffect";
        }
    }

    public IPropertySet Properties
    {
        get
        {
            return _Properties;
        }
    }

    private IPropertySet _Properties;
    public VideoMRCSettings()
    {
        _Properties = (IPropertySet)new PropertySet();
        _Properties.Add("HologramCompositionEnabled", true);
        _Properties.Add("RecordingIndicatorEnabled", false);
        _Properties.Add("VideoStabilizationEnabled", false);
        _Properties.Add("VideoStabilizationBufferLength", 0);
        _Properties.Add("GlobalOpacityCoefficient", 0.9f);
    }
}
```
```csharp
async void CaptureMedia()
{
    var mediaCapture = new MediaCapture();
    await mediaCapture.InitializeAsync(new MediaCaptureInitializationSettings
    {
        PhotoCaptureSource = PhotoCaptureSource.Auto,
        StreamingCaptureMode = StreamingCaptureMode.Video,
        AudioDeviceId = string.Empty,
        SharingMode = MediaCaptureSharingMode.ExclusiveControl,  

    });
    IVideoEffectDefinition ved = new VideoMRCSettings();
    await mediaCapture.AddVideoEffectAsync(ved, MediaStreamType.Photo);

    Windows.Storage.StorageFolder storageFolder = Windows.Storage.ApplicationData.Current.LocalFolder;
    Windows.Storage.StorageFile file = await storageFolder.CreateFileAsync("testfile.jpg", Windows.Storage.CreationCollisionOption.ReplaceExisting);

    
    await mediaCapture.CapturePhotoToStorageFileAsync(ImageEncodingProperties.CreateJpeg(), file);
}
```

**Reply by laultman** [reply]
I like this, does this code call a single capture, it appears to. Have you yet figured out how to stream the photo stills (captures) to storage at some time interval? I am trying to capture photo stills at about 2 per second and guarded by the IMU sensor stabilization. In short, I want to move to a position stop moving my head and have it grab a couple stills. I'm at the thinking about how to do this stage - no code yet.
Love to know if this has already been done.
